### ISSUE TITLE
Are vault keys in the contract open and public?

### ISSUE THREAD

**Issue by smtabatabaie** [dominichackett/dms-hackfs2021#1]
Hi Dominic, very interesting project. I was reviewing it and had also thought about a similar challenge. My question is, is the vault key that you retrieve via 
   
```
function getVaultKey(string calldata vault,string calldata trustee) view external  returns(string memory)
{
	   require(vaults[vault].locked == false, "Vault is locked");
	   
	   return(vaults[vault].trustees[trustee].key);
}
```

public for everyone to see on the contract? because if so, what can prevent the trustee from browsing the contract and blocks and retrieving it before the vault is unlocked? would really appreciate if you can elaborate on this.
Thanks again

**Comment by dominichackett** [comment]
Yes they can retrieve the key but they will have to decrypt it using their private key from ceramic. This is their private key for private messaging.  I don't think they have direct access to this private key in ceramic.   I will have to review the code to confirm this.
